src/components/WorkspaceSwitcher/WorkspaceSwitcher.tsx

```diff
@@ -0,0 +1,373 @@
+import RecentlyUsedWorkspace from './RecentlyUsedWorkspace';
+import RecentWorkspacesComponent from './RecentWorkspacesComponent';
+import FindPathForWorkspaces from './RecentWorkspacesManager';
+import { TreeViewWorkspaceItem, instanceOfTreeViewWorkspaceItem } from './TreeViewWorkspaceItem';
+import Workspace from './Workspace';
+import WorkspaceMenuToggle from './WorkspaceMenuToggle';
+import WorkspaceTreeView from './WorkspaceTreeView';
+import BuildWorkspaceTree from './WorskpaceTreeBuilder';
+
+/**
+ * Defines the statuses we want to update in an atomic fashion when fetching
+ * the workspaces from RBAC.
+ */
+interface CombinedRBACFetchingStatus {
+  isCurrentlyFetchingWorkspacesFromRBAC: boolean;
+  didFetchingWorkspacesFromRBACFail: boolean;
+}
+
+/**
+ * An atom that helps updating the statuses in an atomic fashion when fetching
+ * the worksapces from RBAC.
+ */
+const RBACRestApiStatusWriteAtom = atom(null, (_, set, arg: CombinedRBACFetchingStatus) => {
+  set(isFetchingWorkspacesFromRBAC, arg.isCurrentlyFetchingWorkspacesFromRBAC);
+  set(isFetchingWorkspacesFromRBACError, arg.didFetchingWorkspacesFromRBACFail);
+});
+
+/**
+ * Fetches the principal's workspaces from RBAC.
+ * @returns the promise of the HTTP call.
+ */
+const fetchWorkspacesFromRBAC = (): Promise<AxiosResponse<RBACListWorkspacesResponse>> => {
+  return axios.get<RBACListWorkspacesResponse>('/api/rbac/v2/workspaces/');
+};
+
+/**
+ * Defines the statuses we want to update in an atomic fashion when fecthing
+ * the recently used workspaces from Chrome.
+ */
+interface CombinedChromeFetchingStatus {
+  isCurrentlyFetchingRecentlyUsedWorkspacesFromChrome: boolean;
+  didFetchingRecentlyUsedWorkspacesFromChromeFail: boolean;
+}
+
+/**
+ * An atom that helps updating the statuses in an atomic fashion when fetching
+ * the recently used workspaces from Chrome.
+ */
+const ChromeRestApiStatusWriteAtom = atom(null, (_, set, arg: CombinedChromeFetchingStatus) => {
+  set(isFecthingRecentlyUsedWorkspaces, arg.isCurrentlyFetchingRecentlyUsedWorkspacesFromChrome);
+  set(isFecthingRecentlyUsedWorkspacesError, arg.didFetchingRecentlyUsedWorkspacesFromChromeFail);
+});
+
+/**
+ * Fetches the principal's recently used workspaces from Chrome.
+ * @returns the promise of the HTTP call.
+ */
+const fetchRecentlyUsedWorkspacesFromChrome = (): Promise<AxiosResponse<RecentlySavedWorkspacesResponse>> => {
+  return axios.get<RecentlySavedWorkspacesResponse>('/api/chrome-service/v1/recently-used-workspaces');
+};
+
+const WorkspaceSwitcher = () => {
+  const myFlaggy = useFlag('platform.chrome.workspace-switcher');
+  console.log(`My flag ${myFlaggy}`);
+
+  // State for when the menu toggle is expanded.
+  const [isMenuContainerOpen, setMenuContainerOpen] = useAtom<boolean>(isWorkspacesMenuExpanded);
+
+  // State for the selected tab.
+  const [activeTabKey, setActiveTabKey] = React.useState<string | number>(0);
```

Comment:
Why is it a number or a string? Can we have jus one or another?

Reply:
It's because the `Tabs` component's `onSelect` function is defined as `(event: React.MouseEvent<HTMLElement, MouseEvent>, eventKey: number | string) => void`, so we can't really use just a number there :disappointed:

Reply:
Got it 👍